R/create_obi_cohort.R

```diff
@@ -11,155 +11,281 @@
+            "82058",
+            "82028",
+            "63031",
+            "70001",
+            "25005",
+            "56002",
+            "33001",
+            "63009",
+            "81005",
+            "41010",
+            "41005"
+          ),
+          1,
+          0
+        ),
+        g1_color_2023 = case_when(
+          mdhhs_id %in% c("39002",
+                          "63009",
+                          "41010") ~ "green",
+          mdhhs_id %in% c("82514",
+                          "63031",
+                          "81005") ~ "yellow",
+          mdhhs_id %in% c(
+            "25006",
+            "82053",
+            "63015",
+            "73008",
+            "82058",
+            "82028",
+            "70001",
+            "25005",
+            "56002",
+            "33001",
+            "41005"
+          ) ~ "red",
+          TRUE ~ as.character(NA)
+        ),
+        select_g1_site_2023 = ifelse(
+          mdhhs_id %in% c(
+            "25006",
+            "82053",
+            "63015",
+            "73008",
+            "82058",
+            "82028",
+            "70001",
+            "25005",
+            "56002",
+            "33001",
+            "41005"
+          ),
+          1,
+          0
+        ),
+        g1_site_2024 = ifelse(
+          mdhhs_id %in% c(
+            "25006",
+            "63019",
+            "63029",
+            "82053",
+            "13013",
+            "63023",
+            "82514",
+            "63018",
+            "82537",
+            "82535",
+            "63015",
+            "82058",
+            "82024",
+            "38009",
+            "50014",
+            "70001",
+            "33002",
+            "74004",
+            "81006",
+            "28002",
+            "33001"
+          ),
+          1,
+          0
         )
-      ),
-      g1_site_2023 = ifelse(
-        mdhhs_id %in% c(
-          "25006",
-          "82053",
-          "39002",
-          "82514",
-          "63015",
-          "73008",
-          "82058",
-          "82028",
-          "63031",
-          "70001",
-          "25005",
-          "56002",
-          "33001",
-          "63009",
-          "81005",
-          "41010",
-          "41005"
+      )
+    
+    if (limit_to_locked == T) {
+      df1 |> filter(
+        flg_complete == 1,
+        case_locked == 1,
+        infant_dob_dt >= lubridate::ymd_hms("2020-01-01 00:00:00")
+      )
+    }
+    
+    else if (limit_to_locked == F) {
+      df1 |> filter(flg_complete == 1,
+                    infant_dob_dt >= lubridate::ymd_hms("2020-01-01 00:00:00"))
+    }
+    
+  } else{
+    df1 = df |>
+      mutate(
+        infant_dob_dt_hms = lubridate::ymd_hms(infant_dob_dt),
+        infant_dob_dt = lubridate::as_date(infant_dob_dt_hms),
+        #case locks at MIDNIGHT AFTER THIS DATE
+        case_lock_dt = infant_dob_dt_hms + days(91),
+        case_locked = ifelse(as_date(case_lock_dt) < lubridate::today(), 1, 0),

```

Comment:
use the existing derived var locked_90days_flg instead.

I would try to avoid doing data processing in this function. If data processing needed, we should plan to move them to the nightly export data processing repo.

see dictionary at https://docs.google.com/spreadsheets/d/1r-3q5KoxdkKh7ccaDl5IAH4F1O3_FgKrp4O-VNEzVPA/edit#gid=0